doc/TorEvents.markdown

```diff
@@ -0,0 +1,301 @@
+# Tor Events used by PrivCount
+
+PrivCount obtains its Tor usage data via Tor Control Events. These events have
+been implemented specifically for PrivCount. PrivCount also uses some other
+Controller interfaces.
+
+## PrivCount Event Overview
+
+### Initialisation
+
+#### PrivCount authenticates to the Control Port (PROTOCOLINFO, AUTHENTICATE),
+     using one of the following methods:
+     * SAFECOOKIE (AUTHCHALLENGE)
+     * HASHEDPASSWORD
+     * NULL (if neither other method is available, and tor offers NULL
+       authentication)
+
+#### PrivCount discovers some relay information (GETINFO, GETCONF)
+
+### Collection
+
+#### PrivCount waits for the collection round to start
+#### PrivCount turns on the tor PrivCount code (SETCONF EnablePrivCount 1)
+#### PrivCount turns on the tor events required for the collection round
+     (SETEVENTS PRIVCOUNT_...)
+#### Tor starts sending events as they occur, and PrivCount processes these
+     events
+
+### Cleanup
+
+#### PrivCount waits for the collection round to stop
+#### PrivCount turns off EnablePrivCount and all tor events (SETCONF,
+     SETEVENTS)
+
+## PrivCount Event Overview
+
+PrivCount implements the following events, listed in the typical order they
+would occur for a single client connection:
+
+### PRIVCOUNT_DNS_RESOLVED
+
+This event is sent when an exit receives a client request, including:
+* both connect and resolve requests,
+* for DNS names, and IPv4 and IPv6 addresses,
+* whether the DNS name is in Tor's local DNS cache or not,
+* with limited checks on whether the address is a valid DNS name or IP address.
+  (Tor clients and Tor Exits perform some sanity checks before sending an
```

Comment:
Note that if you want this line to be aligned with the line above it (the last item in the bulleted list) when viewed as markdown, you would end the line above it with 2 spaces.

For example:

\* item 1
\* item 2\space\space
more stuff

gives you:

* item 1
* item 2  
more stuff